### PR TITLE
Add FXIOS-7180 [v121] Private mode theme

### PR DESCRIPTION
**Pull request by cyndichin**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7180)

## :bulb: Description
* Add private mode theme colors and removed unused `getThemedImageName` method
* Private colors will default to dark theme colors unless specified in the design system

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: BrowserKit/Sources/Common/Theming/ThemeType.swift
Comment (cyndichin) [inline BrowserKit/Sources/Common/Theming/ThemeType.swift:35]: This is currently not being used and confirmed with design that it is not needed for private mode as it will use dark mode assets.

File: BrowserKit/Sources/Common/Theming/DarkTheme.swift
Comment (lmarceau) [inline BrowserKit/Sources/Common/Theming/DarkTheme.swift:105]: That one confused me. It's placed under the `Layers` section in the [Figma file](https://www.figma.com/file/pEyGeE4KV5ytYHeXMfLcEr/Mobile-Styles?type=design&node-id=1594-15823&mode=design&t=Odj3QUw2Gn4L9Pnh-4), but then the name is `borderToolbarDivider`. Since it was starting with `border` I was searching it under the border section in Figma and not the layer one. Can we double-check if this is indeed a layer, or more a border?

Comment (cyndichin) [reply BrowserKit/Sources/Common/Theming/DarkTheme.swift:105]: It should be under the Borders section, this was a mishap and crystal should be fixing this. There were some other discrepancies I found, which I documented the rest to be addressed in this ticket: https://mozilla-hub.atlassian.net/browse/FXIOS-7778

In regards to how our tokens are ordered, some members of the team chatted in coffee chat today about possibly having it sorted alphabetically. The design team is going to discuss this and I was going to follow up with the team on what we preferred after that conversation.

Comment (cyndichin) [reply BrowserKit/Sources/Common/Theming/DarkTheme.swift:105]: Actually, the Figma file should be updated now! please verify again and let me know if any issues! @lmarceau

Comment (lmarceau) [reply BrowserKit/Sources/Common/Theming/DarkTheme.swift:105]: Awesome thank you!

**Review by lmarceau — APPROVED** [review]
LGTM from my side of things 👍 Will wait for Roux for final approval

**Review by adudenamedruby — APPROVED** [review]

**Pull request merged**
